src/EditorFeatures/Core/ExternalAccess/VSTypeScript/VSTypeScriptInProcLanguageClient.cs

```diff
@@ -57,6 +58,15 @@ public override ServerCapabilities GetCapabilities(ClientCapabilities clientCapa
         serverCapabilities.ProjectContextProvider = true;
 
         serverCapabilities.SupportsDiagnosticRequests = true;
+        serverCapabilities.DiagnosticProvider = new()
+        {
+            SupportsMultipleContextsDiagnostics = true,
+            DiagnosticKinds =
+            [
+                new(PullDiagnosticCategories.Task),
```

Comment:
@dibarbet for eyes.  not sure if Maria also needs to specify other things for the other diagnostics (syntax, semantics, etc.).

Reply:
Just added everything that https://github.com/dotnet/roslyn/blob/1bf23610b41c148e1cf2d4144d0dbd6714b1f145/src/EditorFeatures/Core/LanguageServer/AlwaysActivateInProcLanguageClient.cs#L75 includes.

Reply:
I don't believe the other ones would be necessary, but it probably won't hurt much to have them/.

Reply:
Well before the last commit every diagnostic that was not a TODO didn't show up hahaha so I think we do need them.

Reply:
Oh - I bet if we have any kind at all - the client will only call us for that kind (and not also call us for no kind).  Which II think generally makes sense.  So yeah we either need all of these, or we need to add an all kind.  I'm fine with this as-is.

Reply:
Yep, that's my hypothesis as well.